Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/Reflection/LegacyReflectionBridge.cpp

```diff
@@ -0,0 +1,351 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/Component/ComponentApplicationBus.h>
+#include <AzCore/DOM/DomPath.h>
+#include <AzCore/DOM/DomUtils.h>
+#include <AzCore/Interface/Interface.h>
+#include <AzCore/Name/Name.h>
+#include <AzCore/RTTI/AttributeReader.h>
+#include <AzCore/RTTI/TypeInfo.h>
+#include <AzFramework/DocumentPropertyEditor/PropertyEditorSystemInterface.h>
+#include <AzFramework/DocumentPropertyEditor/Reflection/LegacyReflectionBridge.h>
+
+namespace AZ::Reflection
+{
+    namespace DescriptorAttributes
+    {
+        Name Handler = Name::FromStringLiteral("Handler");
+        Name Label = Name::FromStringLiteral("Label");
+        Name SerializedPath = Name::FromStringLiteral("SerializedPath");
+    } // namespace DescriptorAttributes
+
+    namespace LegacyReflectionInternal
+    {
+        struct InstanceVisitor
+            : IObjectAccess
+            , IAttributes
+        {
+            IReadWrite* m_visitor;
+            SerializeContext* m_serializeContext;
+
+            struct AttributeData
+            {
+                // Group from the attribute metadata (generally empty with Serialize/EditContext data)
+                Name m_group;
+                // Name of the attribute
+                Name m_name;
+                // DOM value of the attribute - currently only primitive attributes are supported,
+                // but other types may later be supported via opaque values
+                Dom::Value m_value;
+            };
+
+            struct StackEntry
+            {
+                void* m_instance;
+                AZ::TypeId m_typeId;
+                const SerializeContext::ClassData* m_classData = nullptr;
+                const SerializeContext::ClassElement* m_classElement = nullptr;
+                AZStd::vector<AttributeData> m_cachedAttributes;
+                AZStd::string m_path;
+            };
+            AZStd::stack<StackEntry> m_stack;
+
+            using HandlerCallback = AZStd::function<bool()>;
+            AZStd::unordered_map<AZ::TypeId, HandlerCallback> m_handlers;
+
+            virtual ~InstanceVisitor() = default;
+
+            InstanceVisitor(IReadWrite* visitor, void* instance, const AZ::TypeId& typeId, SerializeContext* serializeContext)
+                : m_visitor(visitor)
+                , m_serializeContext(serializeContext)
+            {
+                m_stack.push({ instance, typeId });
+                RegisterPrimitiveHandlers<bool, uint8_t, uint16_t, uint32_t, uint64_t, int8_t, int16_t, int32_t, int64_t, float, double>();
```

Comment:
You should switch `uint64_t` and `int64_t` to be `AZ::u64` and `AZ::s64` since those types are always `unsigned long long` and `long long` across all platforms

While the C standard typedefs of `uint64_t` and `int64_t` are 64-bits across all platforms, they are not the same underlying type(`long` on Linux, Android, `long long` on Windows, Mac, iOS) which will cause issues if they are serialized out.

Reply:
I don't think there's a potential issue with serialization here (we're always just comparing against the type ID of the actual in-memory instance which will be per-platform correct) but I've updated the aliases here just in case the C standard typedefs don't end up including `long long` or anything like that, which would lead to unhandled property types.